### PR TITLE
multi: remove Litecoin

### PR DESCRIPTION
**Pull request by ellemouton**
This PR removes the ability to configure LND with a Litecoin backend.

### CODE REVIEW

File: chainreg/chainregistry.go [hunk 1]
```diff
@@ -256,46 +219,22 @@ func GenDefaultBtcConstraints() channeldb.ChannelConstraints {
...
+		MinHTLCOut:    cfg.Bitcoin.MinHTLCOut,
+		BaseFee:       cfg.Bitcoin.BaseFee,
+		FeeRate:       cfg.Bitcoin.FeeRate,
+		TimeLockDelta: cfg.Bitcoin.TimeLockDelta,
 	}
+	cc.MinHtlcIn = cfg.Bitcoin.MinHTLCIn
+	cc.FeeEstimator = chainfee.NewStaticEstimator(
+		DefaultBitcoinStaticFeePerKW,
+		DefaultBitcoinStaticMinRelayFeeRate,
+	)
```
Comment (morehouse) [inline chainreg/chainregistry.go:237]: We could simplify by folding this all into one statement:
```go
cc := &PartialChainControl{
  Cfg: cfg,
  RoutingPolicy = htlcswitch.ForwardingPolicy{
    MinHTLCOut:    cfg.Bitcoin.MinHTLCOut,
    BaseFee:       cfg.Bitcoin.BaseFee,
    FeeRate:       cfg.Bitcoin.FeeRate,
    TimeLockDelta: cfg.Bitcoin.TimeLockDelta,
  },
  MinHtlcIn = cfg.Bitcoin.MinHTLCIn,
  FeeEstimator = chainfee.NewStaticEstimator(
    DefaultBitcoinStaticFeePerKW,
    DefaultBitcoinStaticMinRelayFeeRate,
  ),
}
```

We can also move the `cc` definition closer to the switch where it gets used.

File: chainreg/chainregistry.go [hunk 2]
```diff
@@ -256,46 +219,22 @@ func GenDefaultBtcConstraints() channeldb.ChannelConstraints {
 //
 //nolint:lll
 func NewPartialChainControl(cfg *Config) (*PartialChainControl, func(), error) {
-	// Set the RPC config from the "home" chain. Multi-chain isn't yet
-	// active, so we'll restrict usage to a particular chain for now.
 	homeChainConfig := cfg.Bitcoin
```
Comment (morehouse) [inline chainreg/chainregistry.go:222]: We could probably get rid of this temporary variable entirely.

**Review by morehouse — APPROVED** [review]

File: chainreg/chainregistry.go (hunk 1 above)
Comment (ellemouton) [reply chainreg/chainregistry.go]: nice - done 👍

File: chainreg/chainregistry.go (hunk 2 above)
Comment (ellemouton) [reply chainreg/chainregistry.go]: removed 👍

**Comment by ellemouton** [comment]
Thanks for the comments @morehouse  & @feelancer21 ! Updated 🚀

**Comment by ellemouton** [comment]
@feelancer21  - any ideas why the sample conf script fails? it passes locally for me

**Comment by feelancer21** [comment]
> @feelancer21 - any ideas why the sample conf script fails? it passes locally for me

Yes and No. The error is 
`Option tlsextradomain: defaults don't match - sample-lnd.conf:                 '', lnd: 'false'`

The option already gave me a headache during the implementation because it is mentioned in the help text too. Must be related to tmux. But I have no idea why it occurs now again. I'll look at it in detail later.

File: config.go [hunk 3]
```diff
@@ -1159,252 +1123,147 @@ func ValidateConfig(cfg Config, interceptor signal.Interceptor, fileParser,
...
-
-		switch cfg.Litecoin.Node {
-		case "ltcd":
-			err := parseRPCParams(
-				cfg.Litecoin, cfg.LtcdMode,
-				chainreg.LitecoinChain, cfg.ActiveNetParams,
+	// For backwards compatibility, the cfg.Bitcoin.Active option is still
+	// exposed to the user, but it now must be set to true since no other
+	// chains are supported.
+	if !cfg.Bitcoin.Active {
```
Comment (yyforyongyu) [inline config.go:1137]: nit: maybe we should mark it deprecated?

File: docs/release-notes/release-notes-0.17.0.md [hunk 4]
```diff
@@ -154,6 +154,7 @@
   latest version of `google.golang.org/protobuf` instead of the deprecated
   `github.com/golang/protobuf/jsonpb`
   module](https://github.com/lightningnetwork/lnd/pull/7659).
+* [Remove Litecoin code](https://github.com/lightningnetwork/lnd/pull/7867)
```
Comment (yyforyongyu) [inline docs/release-notes/release-notes-0.17.0.md:157]: Should be 18 now, and we should also mention the deprecated config fields.

File: htlcswitch/hop/payload.go [hunk 5]
```diff
@@ -106,7 +106,6 @@ func NewLegacyPayload(f *sphinx.HopData) *Payload {
 
 	return &Payload{
 		FwdInfo: ForwardingInfo{
-			Network:         BitcoinNetwork,
```
Comment (yyforyongyu) [inline htlcswitch/hop/payload.go:109]: Just realized this is a hard-coded value😂, and the `Realm` field for the legacy hop is never used...

**Review by yyforyongyu — COMMENTED** [review]
Happy removing🥳🤩
Need to update the release notes, otherwise good to go🏖

File: config.go (hunk 3 above)
Comment (ellemouton) [reply config.go:1137]: mark it deprecated and then just ignore the actual value? or still read the actual value for now? 

Ie, do we mark it deprecated but then still require that they set it?

Comment (yyforyongyu) [reply config.go:1137]: good q...I think we can ignore the value

Comment (ellemouton) [reply config.go:1137]: cool - done!

File: docs/release-notes/release-notes-0.17.0.md (hunk 4 above)
Comment (ellemouton) [reply docs/release-notes/release-notes-0.17.0.md]: done :)

**Review by yyforyongyu — APPROVED** [review]
LGTM🌊

**Review by GeorgeTsagk — APPROVED** [review]

**Comment by guggero** [comment]
Needs a rebase.

File: rpcserver.go [hunk 6]
```diff
@@ -2955,12 +2955,11 @@ func (r *rpcServer) GetInfo(_ context.Context,
...
-	for i, chain := range r.cfg.registeredChains.ActiveChains() {
-		activeChains[i] = &lnrpc.Chain{
-			Chain:   chain.String(),
+	activeChains := []*lnrpc.Chain{
+		{
+			Chain:   bitcoinChainName,
 			Network: network,
-		}
+		},
 	}
```
Comment (ProofOfKeags) [inline rpcserver.go:2963]: I'm not sure what the process is for deprecated values but do we need a TODO to remove this when we actually strip this value from a future GetInfoResponse?

File: zpay32/invoice_test.go [hunk 7]
```diff
@@ -17,9 +17,7 @@ import (
 	"github.com/btcsuite/btcd/btcutil"
 	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
-	"github.com/btcsuite/btcd/wire"
```
Comment (ProofOfKeags) [inline zpay32/invoice_test.go:20]: was this ever needed? Seems unrelated to litecoin removal but I could be missing something.

**Review by ProofOfKeags — COMMENTED** [review]
All nits here. Feel free to disregard all of it, but I wanted to give it a thorough look and mention the things I did see. RIP Litecoin 🪦. Nice job :)

File: zpay32/invoice_test.go (hunk 7 above)
Comment (ellemouton) [reply zpay32/invoice_test.go:20]: you mean the `"github.com/btcsuite/btcd/wire"` import? it was used by the litecoin tests. see below. Golang doesnt allow you to have an import that you dont use unless you explicitly prefix it with `_`

**Review by ellemouton — COMMENTED** [review]
Thanks @ProofOfKeags - rebased and added the suggested changes on at the end as 2 new commits

**Review by ProofOfKeags — APPROVED** [review]
Send it 🚀

**Comment by guggero** [comment]
The parser of the config file format is a bit wonky. So having things in there that don't exist anymore would explain the behavior.

**Comment by theborakompanioni** [comment]
> The parser of the config file format is a bit wonky. So having things in there that don't exist anymore would explain the behavior.

From https://github.com/lightningnetwork/lnd/blob/v0.18.0-beta/config.go#L771-L776:
```
// If it's a parsing related error, then we'll return
// immediately, otherwise we can proceed as possibly the config
// file doesn't exist which is OK.
if _, ok := err.(*flags.IniError); ok {
	return nil, err
}
```

Do I understand correctly that this means it continued as if no config file was present?

**Comment by guggero** [comment]
Would need to look into it in more detail, but I think the `flags.IniError` is only returned if the actual format of the file is wrong (so it cannot be parsed correctly), not when its content is incorrect.

So maybe the PR @michael1011 just put up right now (#8822) fixes the problem?
Would be great if you could add the Litecoin parts back and test if you get a proper error when you run that PR?

**Pull request merged**